Fix `C3DTilesLayer` and `C3DTFeature`

valentinMachado opened the pull request:
## Description
`C3DTFeature` reference the `THREE.Object3D` where they belong.
`computeWorldBox3` is now a method of `C3DTFeature` possible due to the latter modification
Some code clean.
`updateStyle` is call after dispacthing event `ON_TILE_CONTENT_LOADED`

## Motivation and Context
`C3DTFeature` reference the `THREE.Object3D` where they belong.
=> tileContent can have different child with feature in it which was not handle by the previous code
`updateStyle` is call after dispacthing event `ON_TILE_CONTENT_LOADED`
=> when listener of this event influences `C3DTilesLayer` style the user does not have to call an extra `updateStyle` of the layer

waiting this to be approved to rename commits name

jailln reviewed (approved):
Ok for me 👍 thanks !

valentinMachado commented:
Great ! I've made a single commit, and CI is working.

jailln commented on src/Core/3DTiles/C3DTFeature.js:
```diff
@@ -1,23 +1,28 @@
-    constructor(tileId, batchId, groups, info, userData = {}) {
-        /** @type {number} */
-        this.tileId = tileId;
+    constructor(tileId, object3d, batchId, groups, info, userData = {}) {
```
I just realized that this class is exposed in the API. Can you handle the case when users use the old API, log a deprecation message and add a BREAKING CHANGE message in the body of your commit please ?

valentinMachado replied:
Done

valentinMachado committed:
fix(C3DTilesLayer): handle tileContent with several child containing C3DTFeature.
BREAKING CHANGE: C3DTFeature constructor changed from (tileId, batchId, groups, info, userData) to (tileId, batchId, groups, info, userData, object3d)
